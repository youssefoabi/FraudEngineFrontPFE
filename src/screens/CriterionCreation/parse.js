import { format } from 'date-fns';
import { isNil, map, pathOr } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const parse = criterion => {
  if (isNil(criterion)) return null;

  const { id, code, explanation, createdAt } = criterion;
  const eligibleOperators = pathOr([], ['eligibleOperators'], criterion);

  const mapEligibleOperators = allEligibleOperators => {
    if (isNilOrEmpty(allEligibleOperators)) return null;

    return map(eligibleOperator => pathOr('', ['value'], eligibleOperator), allEligibleOperators);
  };

  return {
    id,
    code,
    explanation,
    eligibleOperators: mapEligibleOperators(eligibleOperators),
    createdAt,
    modifiedAt: format(new Date(), 'yyyy-MM-dd'),
  };
};
export default parse;
