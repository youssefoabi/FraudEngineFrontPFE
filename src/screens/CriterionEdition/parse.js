import { isNil, map, pathOr } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const parseCriterion = criterion => {
  if (isNil(criterion)) return null;

  const { id, code, explanation } = criterion;
  const eligibleOperators = pathOr([], ['eligibleOperators'], criterion);

  const mapEligibleOperators = allEligibleOperators => {
    if (isNilOrEmpty(allEligibleOperators)) return null;

    return map(
      eligibleOperator => ({ label: eligibleOperator, value: eligibleOperator }),
      allEligibleOperators,
    );
  };

  return {
    previousId: id,
    id,
    code,
    explanation,
    eligibleOperators: mapEligibleOperators(eligibleOperators),
  };
};
export default parseCriterion;
