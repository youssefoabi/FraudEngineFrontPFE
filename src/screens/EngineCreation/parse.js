import { isNil, map, pathOr, pipe, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

import { FIELDS_BY_ID } from '../../components/SearchCriteriaSettings/components/FieldsBlock/config';

const parse = engine => {
  if (isNil(engine)) return null;

  const {
    id,
    name,
    isEnable,
    scopes,
    inputFields,
    backgroundImages,
    marketingText,
    isMarketingTextEnable,
    searchText,
    engineId,
    createdAt,
    modifiedAt,
  } = engine;

  const logo = {
    urlImageDesktop: 'string',
    urlLinkDesktop: 'string',
    urlImageMobile: 'string',
    urlLinkMobile: 'string',
    order: 1,
    isEnable: true,
    alt: 'string',
    openInNewTab: true,
  };

  const mapParameter = param => {
    if (isNilOrEmpty(param)) return null;

    const { scopeId: scopeParameterId, key: label } = param || {};
    return { ...param, scopeParameterId, label };
  };

  const mapInputFields = allInputFields => {
    if (isNilOrEmpty(allInputFields)) return null;

    return pipe(
      values,
      map(inputField => {
        const type = pathOr(null, ['type', 'value'], inputField);
        const parameters = pathOr([], ['parameters'], inputField);
        const { label } = inputField || {};
        const newParameters = map(mapParameter, parameters);
        return {
          ...(inputField || {}),
          type,
          fieldTypeId: FIELDS_BY_ID[type],
          inputFieldLabel: label,
          parameters: newParameters,
        };
      }),
    )(allInputFields);
  };

  return {
    engineCode: id,
    code: id,
    name,
    isEnable,
    scopes: isNilOrEmpty(scopes) ? [] : values(scopes),
    inputFields: isNilOrEmpty(inputFields) ? [] : mapInputFields(inputFields),
    backgroundImages: isNilOrEmpty(backgroundImages) ? [] : values(backgroundImages),
    marketingText: {
      text: marketingText,
      isEnable: isMarketingTextEnable,
    },
    searchText,
    engineId,
    logo,
    createdAt,
    modifiedAt,
  };
};
export default parse;
