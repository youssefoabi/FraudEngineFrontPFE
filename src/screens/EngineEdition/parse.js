import { addIndex, find, fromPairs, isNil, map, pathOr, pipe, prop } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

import { typesOptions } from '../../components/SearchCriteriaSettings/components/FieldsBlock/config';

const toTimestamp = date => (date ? Number(new Date(date)) : null);

const parseEngine = engine => {
  if (isNil(engine)) return null;

  const {
    id,
    code,
    name,
    isEnable,
    scopes,
    inputFields,
    backgroundImages,
    marketingText,
    searchText,
    createdAt,
    modifiedAt,
  } = engine;

  const { text, isEnable: isMarketingTextEnable } = marketingText || {};

  const mapScopes = allScopes => {
    if (isNilOrEmpty(allScopes)) return null;

    return pipe(
      map(scope => [prop('scopeId', scope), scope]),
      fromPairs,
    )(allScopes);
  };

  const mapInputFields = allInputFields => {
    if (isNilOrEmpty(allInputFields)) return null;
    const mapIndexed = addIndex(map);

    return pipe(
      map(inputField => {
        const type = pathOr(null, ['type'], inputField);
        return { ...(inputField || {}), type: find(option => type === prop('value', option), typesOptions) };
      }),
      mapIndexed((inputField, index) => [index, inputField]),
      fromPairs,
    )(allInputFields);
  };

  const mapBackgroundImages = allBackgroundImages => {
    if (isNilOrEmpty(allBackgroundImages)) return null;

    const mapIndexed = addIndex(map);
    return pipe(
      mapIndexed((backgroundImage, index) => [index, backgroundImage]),
      fromPairs,
    )(allBackgroundImages);
  };

  const firstFieldId = pathOr(null, ['0', 'id'], inputFields);
  const secondFieldId = pathOr(null, ['1', 'id'], inputFields);

  return {
    id: code,
    engineId: id,
    name,
    isEnable,
    scopes: mapScopes(scopes),
    inputFields: mapInputFields(inputFields),
    backgroundImages: mapBackgroundImages(backgroundImages),
    marketingText: text,
    isMarketingTextEnable,
    searchText,
    createdAt,
    modifiedAt: toTimestamp(modifiedAt),
    firstFieldId,
    secondFieldId,
  };
};
export default parseEngine;
