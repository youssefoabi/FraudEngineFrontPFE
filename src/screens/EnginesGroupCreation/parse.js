import { isNil, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const parse = enginesGroup => {
  if (isNil(enginesGroup)) return null;

  const { groupId, groupCode, groupName, isEnable, engines } = enginesGroup;

  return {
    groupId,
    groupCode,
    groupName,
    isEnable,
    engines: isNilOrEmpty(engines) ? [] : values(engines),
  };
};
export default parse;
