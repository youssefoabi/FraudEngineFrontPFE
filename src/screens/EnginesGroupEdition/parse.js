import { fromPairs, isNil, map, pipe, prop } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const toTimestamp = date => (date ? Number(new Date(date)) : null);

const parseEnginesGroup = enginesGroup => {
  if (isNil(enginesGroup)) return null;

  const { groupId, groupCode, groupName, isEnable, engines, createdAt, modifiedAt } = enginesGroup;

  const mapEngines = allEngines => {
    if (isNilOrEmpty(allEngines)) return null;

    return pipe(
      map(engine => [prop('engineId', engine), engine]),
      fromPairs,
    )(allEngines);
  };

  return {
    groupId,
    groupCode,
    groupName,
    isEnable,
    engines: mapEngines(engines),
    createdAt,
    modifiedAt: toTimestamp(modifiedAt),
  };
};
export default parseEnginesGroup;
