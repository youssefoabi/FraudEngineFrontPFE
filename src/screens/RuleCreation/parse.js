import { isNil, map, pathOr, pipe, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

import { FIELDS_BY_ID } from '../../components/SearchCriteriaSettings/components/FieldsBlock/config';

const parse = rule => {
  if (isNil(rule)) return null;

  const {
    id,
    name,
    isEnable,
    scopes,
    inputFields,
    backgroundImages,
    marketingText,
    isMarketingTextEnable,
    searchText,
    ruleId,
    createdAt,
    modifiedAt,
  } = rule;

  const mapParameter = param => {
    if (isNilOrEmpty(param)) return null;

    const { scopeId: scopeParameterId, key: label } = param || {};
    return { ...param, scopeParameterId, label };
  };

  const mapInputFields = allInputFields => {
    if (isNilOrEmpty(allInputFields)) return null;

    return pipe(
      values,
      map(inputField => {
        const type = pathOr(null, ['type', 'value'], inputField);
        const parameters = pathOr([], ['parameters'], inputField);
        const { label } = inputField || {};
        const newParameters = map(mapParameter, parameters);
        return {
          ...(inputField || {}),
          type,
          fieldTypeId: FIELDS_BY_ID[type],
          inputFieldLabel: label,
          parameters: newParameters,
        };
      }),
    )(allInputFields);
  };

  return {
    ruleCode: id,
    code: id,
    name,
    isEnable,
    scopes: isNilOrEmpty(scopes) ? [] : values(scopes),
    inputFields: isNilOrEmpty(inputFields) ? [] : mapInputFields(inputFields),
    backgroundImages: isNilOrEmpty(backgroundImages) ? [] : values(backgroundImages),
    marketingText: {
      text: marketingText,
      isEnable: isMarketingTextEnable,
    },
    searchText,
    ruleId,
    createdAt,
    modifiedAt,
  };
};
export default parse;
