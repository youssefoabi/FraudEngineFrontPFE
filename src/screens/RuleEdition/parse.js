import { isNil, map, pathOr } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const toTimestamp = date => (date ? Number(new Date(date)) : null);

const parseRule = rule => {
  if (isNil(rule)) return null;

  const {
    id,
    name,
    description,
    isValidated,
    isActivated,
    priority,
    validatedBy,
    editedBy,
    comment,
    version,
    editedAction,
    criteriaValues,
    simulations,
    createdAt,
    modifiedAt,
  } = rule;

  const mapCriteriaValues = allCriteriaValues => {
    if (isNilOrEmpty(allCriteriaValues)) return null;

    return map(criteriaValue => {
      const operator = pathOr(null, ['operator'], criteriaValue);
      return { ...(criteriaValue || {}), operator: { label: operator, value: operator } };
    }, allCriteriaValues);
  };

  return {
    id,
    name,
    description,
    isValidated,
    isActivated,
    priority,
    validatedBy,
    editedBy,
    comment,
    version,
    editedAction,
    criteriaValues: mapCriteriaValues(criteriaValues),
    simulations,
    createdAt,

    modifiedAt: toTimestamp(modifiedAt),
  };
};
export default parseRule;
