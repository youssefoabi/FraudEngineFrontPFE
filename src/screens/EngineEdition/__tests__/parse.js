import { itReturnsCorrectResult } from '@cdiscount/jest-utils';

import engineFixture from '../__fixtures__';
import parse from '../parse';

describe('parse', () => {
  describe('engine', () => {
    itReturnsCorrectResult(parse(engineFixture));
  });
  describe('null', () => {
    itReturnsCorrectResult(parse(null));
  });
});
