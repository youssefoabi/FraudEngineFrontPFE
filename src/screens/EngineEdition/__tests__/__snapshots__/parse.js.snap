// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse engine returns correct result 1`] = `
Object {
  "backgroundImages": Object {
    "0": Object {
      "alt": "alternative text",
      "isEnable": true,
      "openInNewTab": true,
      "order": 0,
      "urlImageDesktop": "https://www.leparisien.fr/resizer/CBn4GjacGJ5llevItKUbQYa6zcw=/932x582/arc-anglerfish-eu-central-1-prod-leparisien.s3.amazonaws.com/public/4U4PGCRHAK2F6HLMZCBI4PSDNU.jpg",
      "urlImageMobile": "https://www.leparisien.fr/resizer/CBn4GjacGJ5llevItKUbQYa6zcw=/932x582/arc-anglerfish-eu-central-1-prod-leparisien.s3.amazonaws.com/public/4U4PGCRHAK2F6HLMZCBI4PSDNU.jpg",
      "urlLinkDesktop": "Lien PC",
      "urlLinkMobile": "Lien Mobile",
    },
  },
  "createdAt": "2020-05-13T15:39:07.932Z",
  "engineId": 1,
  "firstFieldId": 0,
  "id": 0,
  "inputFields": Object {
    "0": Object {
      "id": 0,
      "isEnable": true,
      "isMandatory": true,
      "label": "Placeholder Field 1",
      "order": 0,
      "parameters": Array [
        Object {
          "externalCodeId": 4,
          "id": 4,
          "key": "Latitude",
          "scopeId": 2,
        },
      ],
      "type": Object {
        "fieldTypeId": 1,
        "label": "Autocomplétion d'adresse",
        "value": "Address",
      },
    },
    "1": Object {
      "id": 1,
      "isEnable": true,
      "isMandatory": true,
      "label": "Placeholder Field 2",
      "order": 0,
      "parameters": Array [
        Object {
          "externalCodeId": 2,
          "id": 0,
          "key": "cp",
          "scopeId": 1,
        },
        Object {
          "externalCodeId": 4,
          "id": 1,
          "key": "commune",
          "scopeId": 1,
        },
        Object {
          "externalCodeId": 1,
          "id": 3,
          "key": "search",
          "scopeId": 2,
        },
      ],
      "type": Object {
        "fieldTypeId": 1,
        "label": "Autocomplétion d'adresse",
        "value": "Address",
      },
    },
  },
  "isEnable": true,
  "isMarketingTextEnable": true,
  "marketingText": "marketingText",
  "modifiedAt": 1589384347932,
  "name": "name",
  "scopes": Object {
    "1": Object {
      "isEnable": true,
      "name": "gamme active avec un param n'est pas lie...",
      "order": 0,
      "scopeId": 1,
    },
    "2": Object {
      "isEnable": true,
      "name": "gamme active valide",
      "order": 0,
      "scopeId": 2,
    },
    "3": Object {
      "isEnable": false,
      "name": "gamme disactive",
      "order": 0,
      "scopeId": 3,
    },
  },
  "searchText": "searchText",
  "secondFieldId": 1,
}
`;

exports[`parse null returns correct result 1`] = `null`;
